docs/migration-guide/to-15.md

```diff
@@ -1,12 +1,65 @@
 # Migrating to 15.0.0
 
-## Remove support for processors
+This release contains significant and breaking changes that will help us modernize our code so that Stylelint remains free of security issues.
 
-We removed the support for processors. Instead, please use [custom syntaxes](../developer-guide/syntaxes.md).
+## Significant change
 
-## Change of `overrides.extends` behavior
+We deprecated [most of the rules that enforce stylistic conventions](../user-guide/rules.md#deprecated). You should remove these rules from your configuration object.
 
-We changed the `overrides.extends` behavior to merge rather than replace, to make it consistent with the `overrides.plugins`.
+We recommend:
+
+- using a pretty printer, like [Prettier](https://prettier.io/), alongside Stylelint
+- extending [our standard config](https://www.npmjs.com/package/stylelint-config-standard) in your configuration object
+
+Stylelint and pretty printers are complementary tools that work together to help you write consistent and error-free code. To help facilitate this, we've removed the [deprecated rules](../user-guide/rules.md#deprecated) from the latest version of [our standard config](https://www.npmjs.com/package/stylelint-config-standard). You can [extend](../user-guide/configure.md#extends) the config using:
+
+```diff json
+{
++ "extends": ["stylelint-config-standard"],
+  "rules" { .. }
+}
+```
+
+[Our standard config](https://www.npmjs.com/package/stylelint-config-standard) turns on many of the [other rules that enforce conventions](../user-guide/rules.md#enforce-conventions), e.g. most of the [`*-notation`](../user-guide/rules.md#notation), [`*-pattern`](../user-guide/rules.md#pattern) and [`*-quotes`](../user-guide/rules.md#quotes) rules. We recommend adding more of [the rules that enforce conventions](../user-guide/rules.md#enforce-conventions) to your config as many of them will be specific to your needs, e.g. what [units you allow](../../lib/rules/unit-allowed-list/README.md) in your code.
+
+For example, you can:
+
+- [allow, disallow and require many things](../user-guide/rules.md#allowed-disallowed--required)
+- [set limits on things like selectors](../user-guide/rules.md#max--min)
+
+Additionally, you may no longer need to extend [Prettier's Stylelint config](https://www.npmjs.com/package/stylelint-config-prettier) as there should be no conflicting rules:
+
+```diff json
+{
+- "extends": ["stylelint-config-prettier"],
+  "rules" { .. }
+}
+```
+
+If you want to continue using Stylelint to enforce stylistic consistency, you can [migrate the deprecated rules you need to a plugin](../developer-guide/plugins.md). We will remove the rules from Stylelint in the next major release.
```

Comment:
[Note] This is at the bottom as it's not the recommended route. The rules are problematic and a plugin would suffer the same issues:

- old code reliant on stylesearch
- not cleanly working with the AST
- introducing new lint problems
- a confusing number of configuration options